src/lib/csv.pl

```diff
@@ -222,7 +222,8 @@
   ( separator(Opt) ->
     row(Y, Opt)
   ; end_token ->
-    { Y = [] }).
+    { Y = [] }
+  ; { true }).
```

Comment:
I think this should be `{ false }` instead of `{ true }` to preserve the original meaning?

And then it could also simply be: `end_token, { Y = [] }`, without any further `->`.